assets/app/scripts/controllers/createRoute.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateRouteController
+ * @description
+ * # CreateRouteController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateRouteController', function ($filter, $routeParams, $scope, $window, ApplicationGenerator, DataService, Navigate, project) {
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};
+    $scope.renderOptions.hideFilterWidget = true;
+    $scope.projectName = $routeParams.project;
+    $scope.serviceName = $routeParams.service;
+
+    // Prefill route name with the service name.
+    $scope.routing = {
+      name: $scope.serviceName || ""
+    };
+
+    project
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+
+        var updatePortOptions = function(service) {
+          if (!service) {
+            return;
+          }
+
+          $scope.routing.portOptions = _.map(service.spec.ports, function(portMapping) {
+            return {
+              containerPort: portMapping.targetPort,
+              protocol: portMapping.protocol
+            };
+          });
+
+          if ($scope.routing.portOptions.length) {
+            $scope.routing.targetPort = $scope.routing.portOptions[0];
+          }
+        };
+
+        var watches = [],
+            labels = {},
+            orderByDisplayName = $filter('orderByDisplayName');
+        if ($scope.serviceName) {
+          DataService.get("services", $scope.serviceName, context).then(function(service) {
+            updatePortOptions(service);
+            labels = angular.copy(service.metadata.labels);
+          });
+        } else {
+          // Prompt the user for the service.
+          watches.push(DataService.watch("services", context, function(services) {
+            $scope.services = orderByDisplayName(services.by("metadata.name"));
+            if (!$scope.routing.service && $scope.services.length) {
+              $scope.routing.service = $scope.services[0];
+            }
+            $scope.$watch('routing.service', function() {
+              updatePortOptions($scope.routing.service);
+              labels = angular.copy($scope.routing.service.metadata.labels);
+            });
+          }));
+        }
+
+        $scope.createRoute = function() {
+          $scope.disableInputs = true;
+          if ($scope.createRouteForm.$valid) {
+            var serviceName = $scope.serviceName || $scope.routing.service.metadata.name;
+            var route = ApplicationGenerator.createRoute($scope.routing, serviceName, labels);
+            DataService.create('routes', null, route, context)
+              .then(function() { // Success
+                // Return to the previous page
+                $window.history.back();
```

Comment:
i know its unlikely... but what if someone goes directly to the create route page, there may not be anything in the history.  can we check if there is anything in the history and if not navigate to the routes list page?

or maybe we should just always navigate to the individual page for the route we just created